applications/CoSimulationApplication/custom_io/co_sim_EMPIRE_API.h

```diff
@@ -117,26 +119,43 @@ static void SendArray(const std::string& rFileName, const int sizeOfArray, const
     output_file.close();
     MakeFileVisible(rFileName);
 
+    EMPIRE_API_LOG(2) << "Finished sending array" << std::endl;
+
     if (PrintTiming) {
         EMPIRE_API_LOG(0) << "Sending Array \"" << rFileName << "\" took: " << ElapsedSeconds(start_time) << " [sec]" << std::endl;
     }
 }
 
 static void ReceiveArray(const std::string& rFileName, const int sizeOfArray, double *data)
 {
+    EMPIRE_API_LOG(2) << "Attempting to receive array \"" << rFileName << "\" with size: " << sizeOfArray << " ..." << std::endl;
+
     WaitForFile(rFileName);
 
     const auto start_time(std::chrono::steady_clock::now());
 
     std::ifstream input_file(rFileName);
     CheckStream(input_file, rFileName);
 
+    int size_read;
+    input_file >> size_read; // the first number in the file is the size of the array
+
+    if (size_read != sizeOfArray) {
+        std::stringstream err_msg;
+        err_msg << "The received size for array \"" << rFileName << "\" is different from what is expected:";
+        err_msg << "\n    Expected size: " << sizeOfArray;
+        err_msg << "\n    Received size: " << size_read;
+        throw std::runtime_error(err_msg.str());
+    }
+
     for (int i=0; i<sizeOfArray; ++i) {
         input_file >> data[i];
```

Comment:
Does a precision of 12 also apply here ? or are we loosing something here ?

Reply:
This is the reading, I thought it would just take the standard one...
Or do you mean that we should specify a precision also for writing?

Reply:
I mean we can/should also  set precision for ifstream here to be sure 

https://en.cppreference.com/w/cpp/io/manip/setprecision

http://www.cplusplus.com/reference/fstream/ifstream/